app_dart/lib/src/request_handlers/push_build_status_to_github.dart

```diff
@@ -55,37 +37,15 @@ class PushBuildStatusToGithub extends ApiRequestHandler<Body> {
       return Body.empty;
     }
 
-    final String repo = request!.uri.queryParameters[fullNameRepoParam] ?? 'flutter/flutter';
-    final RepositorySlug slug = RepositorySlug.full(repo);
+    final String repository = request!.uri.queryParameters[fullNameRepoParam] ?? 'flutter/flutter';
+    final RepositorySlug slug = RepositorySlug.full(repository);
     final DatastoreService datastore = datastoreProvider(config.db);
     final BuildStatusService buildStatusService = buildStatusServiceProvider(datastore);
 
-    final Commit tipOfTreeCommit = Commit(
-      sha: Config.defaultBranch(slug),
-      repository: slug.fullName,
-    );
-    final CiYaml ciYaml = await scheduler!.getCiYaml(tipOfTreeCommit);
-    List<LuciBuilder> postsubmitBuilders = await scheduler!.getPostSubmitBuilders(ciYaml);
-    // Filter the builders to only those that can block the tree
-    postsubmitBuilders = postsubmitBuilders.where((LuciBuilder builder) => !builder.flaky!).toList();
-    final LuciService luciService = luciServiceProvider(this);
-    final Map<LuciBuilder, List<LuciTask>> luciTasks = await luciService.getRecentTasks(builders: postsubmitBuilders);
-
-    String status = GithubBuildStatusUpdate.statusSuccess;
-
-    await Future.forEach(luciTasks.entries, (MapEntry<LuciBuilder, List<LuciTask>> luciTask) async {
-      final List<LuciTask> tasks = luciTask.value;
-      if (tasks.isEmpty) {
-        log.fine('No tasks returned for builder: ${luciTask.key.name}');
-      }
-      final String latestStatus = await buildStatusService.latestLUCIStatus(tasks);
-      if (status == GithubBuildStatusUpdate.statusSuccess && latestStatus == GithubBuildStatusUpdate.statusFailure) {
-        status = GithubBuildStatusUpdate.statusFailure;
-      }
-    });
-    await _insertBigquery(slug, status, Config.defaultBranch(slug), config);
-    await _updatePRs(slug, status, datastore, Config.defaultBranch(slug));
-    log.fine('All the PRs for $repo have been updated with $status');
+    final BuildStatus status = (await buildStatusService.calculateCumulativeStatus(slug))!;
```

Comment:
I guess the main concern here is the latency. This depends on refresh-chromebot-status API, which sometimes causes ~10 min delay.
As we are implementing pubsub support, I would recommend we hold this change until the pubsub is ready.

Reply:
https://github.com/flutter/flutter/issues/90569 is a P1 because the tree status isn't deterministic.

For flutter/flutter, the latency is <5 minutes (3 minute cron + 2 minute refresh chromebot status), whereas flutter/engine is <3 minutes (3 minute cron + <15 second refresh crhomebot status)

Reply:
Yeah, we have the cron job frequency set up for short, but it usually takes longer.
If you take a look at the logs, the refresh api runs every ~6 min for flutter, and ~3 min for engine, whereas the push-build-status-to-github API runs every ~3min for flutter, and 2min for engine.

I am wondering when we can have the pubsub set up. If it is expected soon, then we can wait for that as the bug is open for a couple of months anyway.

@godofredoc may have some thoughts here.

Reply:
Making the current logic deterministic requires small fixes in the way we sort the results from the buildbucket responses. Something like:

* Get the last 30 commits, sort them by commit and associate builds to commits
* Iterate over the set of tasks starting from the most recent commit all the way to the oldest
* If a task fails or if a task does not contain any results close the tree otherwise keep the tree green

However we can wait for this logic to be implemented when we separate the bot to its own microservice.